lib/tasks/launch_promo.rake

```diff
@@ -0,0 +1,28 @@
+require "#{Rails.root}/app/helpers/promos_helper"
+include PromosHelper
+
+namespace :promo do
+  # Generates the promo tokens and sends emails
+  task :launch_promo => :environment do
+    begin
+      unless promo_running?
+        puts "Promo is not running, check the active_promo_id config var."
+      end
+
+      publishers = Publisher.all
+
+      publishers.find_each do |publisher|
+        PublisherPromoTokenGenerator.new(publisher: publisher)      
+      end
+
+      publishers.find_each do |publisher|
+        PromoMailer.activate_promo_2018q1(publisher).deliver
```

Comment:
this will send an email to all publishers. if we encounter an error in the middle of the task, it would difficult to resume without spamming the publishers. see above